src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -68,46 +68,61 @@ public BufferReader(in ReadOnlySequence<byte> buffer)
         /// <summary>
         /// The unread portion of the <see cref="CurrentSpan"/>.
         /// </summary>
-        public ReadOnlySpan<byte> UnreadSpan
+        public ReadOnlySpan<T> UnreadSpan
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get => CurrentSpanIndex == 0 ? CurrentSpan : CurrentSpan.Slice(CurrentSpanIndex);
         }
 
         /// <summary>
-        /// The total number of bytes processed by the reader.
+        /// The total number of {T}s processed by the reader.
         /// </summary>
-        public int ConsumedBytes { get; private set; }
+        public int ConsumedValues { get; private set; }
 
         /// <summary>
-        /// Peeks at the next byte value without advancing the reader.
+        /// Peeks at the next value without advancing the reader.
         /// </summary>
-        /// <returns>The next byte or -1 if at the end of the buffer.</returns>
+        /// <param name="value">The next value or default if at the end.</param>
+        /// <returns>False if at the end of the reader.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public int Peek() => End ? -1 : CurrentSpan[CurrentSpanIndex];
+        public bool Peek(out T value)
```

Comment:
TryPeek/TryRead?

Reply:
Hmm- probably should have been that way from the start. :)